Lib/test/test_pathlib.py

```diff
@@ -2766,6 +2767,15 @@ def test_walk_many_open_files(self):
                 self.assertEqual(next(it), expected)
             path = path / 'd'
 
+    def test_walk_above_recursion_limit(self):
+        base = pathlib.Path(os_helper.TESTFN, 'deep')
+        path = pathlib.Path(base, *(['d']*50))
+        path.mkdir(parents=True)
+
+        with infinite_recursion(40):
```

Comment:
Now that it was merged in the `os.walk` PR, maybe use `test.support.set_recursion_limit` here? Although in practice the functionality is the same, the name `infinite_recursion` is inappropriate to this use and may confuse the reader regarding the purpose of the test (this test has nothing to do with infinite recursion.)
```suggestion
        with set_recursion_limit(40):
```

Reply:
Thank  you! That was my intention all along to also use `set_recursion_limit` but I wanted to wait until it gets merged to prevent duplicate commits and possible merge conflicts that could arise if I did anything slightly differently.